src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -262,6 +263,9 @@ private static bool ShouldVersionOldDocument(DocumentFlags flags, BlittableJsonR
             if (flags.Contain(DocumentFlags.Resolved))
                 return false; // we already versioned it with the a conflicted flag
 
+            if (nonPersistentDocumentFlags.Contain(NonPersistentDocumentFlags.ForceRevisionCreation))
+                return false; // no need to create 2 identical revisions
```

Comment:
If I understand this correctly we want to version only the old document, so this should return `true`, no?